aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -315,6 +315,31 @@ struct NormTwoOps {
 #endif
 };
 
+template <typename acc_t>
+struct NanSumOps {
+  inline C10_DEVICE acc_t reduce(acc_t a, acc_t b, int64_t /*idx*/) const {
+    return a + (at::_isnan(b) ? acc_t{0} : b);
+  }
+
+  inline C10_DEVICE acc_t combine(acc_t a, acc_t b) const {
+    return  a + b;
```

Comment:
How do we know that a and b can't be nan at this point?

Reply:
https://github.com/pytorch/pytorch/blob/ebd869153c6adb37507d2ecb6a9fe3fd495fbb6e/aten/src/ATen/native/cpu/Reduce.h#L159-L164